src/Assets/PaymentMethodAssets.php

```diff
@@ -60,38 +51,45 @@ protected function init() {
 	}
 
 	/**
-	 * Registered payment method scripts for usage.
+	 * Register payment method scripts for usage.
 	 */
 	public function register_payment_method_scripts() {
 		$payment_methods = $this->payment_method_registry->get_all_registered();
-
 		foreach ( $payment_methods as $payment_method ) {
-			$scripts = array_filter( (array) $payment_method->scripts );
-
-			if ( ! empty( $scripts ) ) {
-				foreach ( $scripts as $script ) {
-					$this->asset_api->register_script(
-						$script['handle'],
-						$script['src'],
-						$script['deps']
-					);
-					wp_enqueue_script( $script['handle'] );
+			$script_handles       = $payment_method->get_payment_method_script_handles();
+			$admin_script_handles = $payment_method->get_payment_method_script_handles_for_admin();
+			if ( is_admin() && ! empty( $admin_script_handles ) ) {
+				foreach ( $admin_script_handles as $script_handle ) {
+					wp_enqueue_script( $script_handle );
+				}
+			}
+			if ( ! is_admin() && ! empty( $script_handles ) ) {
+				foreach ( $script_handles as $script_handle ) {
+					wp_enqueue_script( $script_handle );
 				}
 			}
 		}
 	}
 
+	/**
+	 * Register payment method styles for usage.
+	 */
+	public function register_payment_method_styles() {
+		// @todo not sure where/how we should enqueue these yet. They can't
+		// be enqueued in the block render function.
+	}
+
```

Comment:
We'll have to decide how we'll enqueue styles for the payment methods extensions (if they have any). We _might_ be able to use the `enqueue_block_assets` action?

Reply:
I think styles need to be global due to needing including in header so we don't need to enqueue them for extensions.

Thats why its important we have default styles in place for common elements that gateways can utilise where possible.